pkg/provisioner/crdprovisioner.go

```diff
@@ -592,7 +594,8 @@ func (c *CrdProvisioner) PublishVolume(
 	}
 
 	// detach replica volume if volume's maxShares have been fully saturated.
-	if azVolume.Spec.MaxMountReplicaCount > 0 {
+	// need to validate these conditions only when mountReplicas is enabled.
+	if c.mountReplicasEnabled && azVolume.Spec.MaxMountReplicaCount > 0 {
```

Comment:
here too checking `azVolume.Spec.MaxMountReplicaCount` should be enough

Reply:
fixed